htdocs/discover_tab.js

```diff
@@ -0,0 +1,58 @@
+var discover = function () {
+
+    /* "Model-Model-View-Controller"
+     .. or something ..
+     there are two models: the local state and the global rcs+github
+     the local model is private data of this class
+    */
+
+    // local model (all caches of stuff stored in RCS/github)
+    var notebook_info_ = {}, // all notebooks we are aware of
+        num_stars_ = {}, // number of stars for all known notebooks
+        fork_count_ = {};
+
+    function clean_r(obj) { 
+        delete obj.r_attributes; 
+        delete obj.r_type; 
+        return obj; 
+    };
+
+    function load_everything(){
+        var allnotebooks = rcloud.config.get_all_notebook_info();
+
+        var promises = [allnotebooks.then(function(notebooks) {
+            // populate notebook infos
+            notebook_info_ = clean_r(notebooks.infos);
+        }),
+        allnotebooks.then(function(notebooks) {
+            // populate star counts
+            num_stars_ = clean_r(notebooks.stars);
+        }),
+        allnotebooks.then(function(notebooks) {
+            // populate fork counts
+            fork_count_ = clean_r(notebooks.forks);
+        })];
```

Comment:
oh, actually on further inspection, it can be just one callback, since none of the `then` functions are asynchronous themselves. but this code will probably change and look more like (but not exactly like) editor_tab's `get_notebooks_by_user`